### PR TITLE
Add ensemble support to ModelConfig

### PR DESCRIPTION
**Pull request by nv-braf**
Adds new methods to check if the model's platform is ensemble and to change the submodel variant names.

Note: This is merging on to the ensemble branch, not main

### CODE REVIEW

File: model_analyzer/triton/model/model_config.py [hunk 1]
```diff
@@ -258,14 +258,56 @@ def set_cpu_only(self, cpu_only):
...
+        -------
+            List[str]: Sub-model names
+        """
+
+        if not self.is_ensemble():
+            return None
+
+        submodels = [
+            model['modelName']
+            for model in self.to_dict()['ensembleScheduling']['step']
```
Comment (tgerdesnv) [inline model_analyzer/triton/model/model_config.py:291]: Should we check the subkeys before indexing in? Seems to assume everything is in a correct state

File: model_analyzer/triton/model/model_config.py [hunk 2]
```diff
@@ -258,14 +258,56 @@ def set_cpu_only(self, cpu_only):
...
+        return submodels
+
+    def set_submodel_variant_name(self, submodel_name: str,
+                                  variant_name: str) -> None:
+        """
+        Replaces the Ensembles submodel's name with the variant name
+        """
+
+        if not self.is_ensemble():
+            return
```
Comment (tgerdesnv) [inline model_analyzer/triton/model/model_config.py:302]: Can we error out here? that would mean bad code and I don't want to silently do that

File: model_analyzer/triton/model/model_config.py [hunk 3]
```diff
@@ -258,14 +258,56 @@ def set_cpu_only(self, cpu_only):
...
+
+        model_config_dict = self.to_dict()
+        for submodel in model_config_dict['ensembleScheduling']['step']:
+            if submodel['modelName'] == submodel_name:
+                submodel['modelName'] = variant_name
+
+        self._model_config = self.from_dict(model_config_dict)._model_config
```
Comment (tgerdesnv) [inline model_analyzer/triton/model/model_config.py:309]: I don't understand this line. Is the only way to update this object to dump it to a dict, modify it, and then import it back from dict?

File: model_analyzer/triton/model/model_config.py (hunk 2 above)
Comment (nv-braf) [reply model_analyzer/triton/model/model_config.py]: Changed this to raise an TMA exception.

Comment (tgerdesnv) [reply model_analyzer/triton/model/model_config.py]: Add a test to protect that and I think this is ready to be merged. (create an ensemble config that is missing 'step', and expect assert to be raised)

**Comment by nv-braf** [comment]
Added tests to protect missing protobuf sections. Please review

**Pull request merged**